var/spack/repos/builtin/packages/mivisionx/package.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Mivisionx(CMakePackage):
+    """MIVisionX toolkit is a set of comprehensive computer
+    vision and machine intelligence libraries, utilities, and
+    applications bundled into a single toolkit."""
+
+    homepage = "https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX"
+    url      = "https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/archive/rocm-3.8.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+
+    def url_for_version(self, version):
+        if version == Version('1.7'):
+            return "https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/archive/1.7.tar.gz"
+
+        url = "https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/archive/rocm-{0}.tar.gz"
+        return url.format(version)
+
+    version('3.8.0', sha256='4e177a9b5dcae671d6ea9f0686cf5f73fcd1e3feb3c50425c8c6d43ac5d77feb')
+    version('3.7.0', sha256='3ce13c6449739c653139fc121411d94eaa9d764d3d339c4c78fab4b8aa199965')
+    version('1.7', sha256='ff77142fd4d4a93136fd0ac17348861f10e8f5d5f656fa9dacee08d8fcd2b1d8')
+
+    variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
+
+    def flag_handler(self, name, flags):
+        spec = self.spec
+        protobuf = spec['protobuf'].prefix.include
+        if name == 'cxxflags':
+            flags.append('-I{0}'.format(protobuf))
+        return (flags, None, None)
+
+    depends_on('ffmpeg@4.1.1', type='build')
+    depends_on('protobuf@3.5.0', type='build')
+    depends_on('opencv@3.4.6~opencl~opencl_svm~openclamdfft~openclamdblas~python', type='build')
```

Comment:
Are these requirements or preferences? `depends_on` only supports hard requirements. If you don't want to build these variants by default, you can override the defaults in `packages.yaml`.

Reply:
we don't have tagging for ROCm 3.5. The available version during ROCm 3.5 was 1.7. If ROCm 3.5 support is not required for mivisionx, yes we can remove version 1.7.

Reply:
@adamjstewart yes above dependency is hard requirement for mivisionx package & version 1.7 is needed for ROCm 3.5.

Reply:
So it won't build at all if someone tries to use OpenCV 3.4.5 or 3.4.7? Or if someone tries to use OpenCV +opencl or +python?

Reply:
@adamjstewart Able to build with opencv@3.4.6.